src/tls.c

```diff
@@ -860,12 +896,16 @@ static INLINE word16 TLSX_ToSemaphore(word16 type)
 
 /** Checks if a specific light (tls extension) is not set in the semaphore. */
 #define IS_OFF(semaphore, light) \
-    ((semaphore)[(light) / 8] ^  (byte) (0x01 << ((light) % 8)))
+    (!(((semaphore)[(light) / 8] &  (byte) (0x01 << ((light) % 8)))))
```

Comment:
When XOR:
0 ^ 0 = 0
0 ^ 1 = 1
1 ^ 0 = 1
1 ^ 1 = 0
Therefore if any other bits (the ones not being checked) are 1 then the result is non-zero or TRUE.
The code with an AND will mask out the bit to be checked.